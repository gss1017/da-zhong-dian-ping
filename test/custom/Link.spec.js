import React from 'react';
import Link from '../../src/pages/custom/Link';
import renderer from 'react-test-renderer';

// 快照测试
describe('link.js', () => {
    test('Link changes then class when hovered', () => {
        const component = renderer.create(
            <Link page="/b">走你</Link>
        );
        // react 组件渲染
        let tree = component.toJSON();
        // 将渲染结果与快照作匹配
        expect(tree).toMatchSnapshot();

        tree.props.onMouseEnter();
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.props.className).toBe('hovered');

        tree.props.onMouseLeave();
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    // 覆盖组件33代码
    test('Link changes then class when hovered no href', () => {
        const component = renderer.create(
            <Link>走你</Link>
        );
        // react 组件渲染
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
