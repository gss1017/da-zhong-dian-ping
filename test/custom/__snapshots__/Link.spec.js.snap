// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link.js Link changes then class when hovered 1`] = `
<a
  className="normal"
  href="/b"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  走你
</a>
`;

exports[`link.js Link changes then class when hovered 2`] = `
<a
  className="hovered"
  href="/b"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  走你
</a>
`;

exports[`link.js Link changes then class when hovered 3`] = `
<a
  className="normal"
  href="/b"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  走你
</a>
`;

exports[`link.js Link changes then class when hovered no href 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  走你
</a>
`;
